jerry-core/jerryscript-debugger.c

```diff
@@ -0,0 +1,94 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jcontext.h"
+#include "jerryscript-debugger.h"
+#include "jerry-debugger.h"
+
+/**
+ * Checks whether the debugger is connected.
+ *
+ * @return true - if the debugger is connected
+ *         false - otherwise
+ */
+bool
+jerry_debugger_is_connected (void)
+{
+#ifdef JERRY_DEBUGGER
+  return JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED;
+#else
+  return false;
+#endif /* JERRY_DEBUGGER */
+} /* jerry_debugger_is_connected */
+
+/**
+ * Stop execution at the next available breakpoint.
+ */
+void
+jerry_debugger_stop (void)
+{
+#ifdef JERRY_DEBUGGER
+  if ((JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED)
+      && !(JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_BREAKPOINT_MODE))
+  {
+    JERRY_CONTEXT (debugger_flags) = (uint8_t) (JERRY_CONTEXT (debugger_flags) | JERRY_DEBUGGER_VM_STOP);
+    JERRY_CONTEXT (debugger_stop_context) = NULL;
+  }
+#endif /* JERRY_DEBUGGER */
+} /* jerry_debugger_stop */
+
+/**
+ * Continue execution.
+ */
+void
+jerry_debugger_continue (void)
+{
+#ifdef JERRY_DEBUGGER
+  if ((JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED)
+      && !(JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_BREAKPOINT_MODE))
+  {
+    JERRY_CONTEXT (debugger_flags) = (uint8_t) (JERRY_CONTEXT (debugger_flags) & ~JERRY_DEBUGGER_VM_STOP);
+    JERRY_CONTEXT (debugger_stop_context) = NULL;
+  }
+#endif /* JERRY_DEBUGGER */
+} /* jerry_debugger_continue */
+
+/**
+ * The engine is not allowed to stop at breakpoints.
+ */
+void
+jerry_debugger_disable_stop_at_breakpoint (void)
+{
+#ifdef JERRY_DEBUGGER
+  if (JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED)
+  {
+    JERRY_CONTEXT (debugger_flags) = (uint8_t) (JERRY_CONTEXT (debugger_flags) | JERRY_DEBUGGER_VM_IGNORE);
+  }
+#endif /* JERRY_DEBUGGER */
+} /* jerry_debugger_disable_stop_at_breakpoint */
+
+/**
+ * The engine is allowed to stop at breakpoints.
+ */
+void
+jerry_debugger_enable_stop_at_breakpoint (void)
```

Comment:
I can deal with both solution, but IMHO it is unnecessary to merge in this case, because we (probably) won't extend the enum. If we merge it than I'd use a bool argument rather than enum.

`void jerry_debugger_set_enable_stop_at_breakpoint (bool is_enable)`